crates/rpc-client/src/builtin.rs

```diff
@@ -20,6 +20,62 @@ pub enum BuiltInConnectionString {
     Ipc(std::path::PathBuf),
 }
 
+/// Configuration for connecting to built-in transports.
+///
+/// Provides a flexible way to configure various aspects of the connection,
+/// including authentication, retry behavior, and transport-specific settings.
+#[derive(Clone, Debug, Default)]
+#[non_exhaustive]
+pub struct ConnectionConfig {
```

Comment:
this def is out of order and in between a struct and impl block

let moves this below all things BuiltInConnectionString